plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/KickCommand.java

```diff
@@ -48,33 +46,15 @@ public KickCommand() {
     // - kick <[player]> "reason:Because I can."
     // -->
 
-    @Override
-    public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {
-        for (Argument arg : scriptEntry) {
-            if (arg.matchesPrefix("reason")) {
-                scriptEntry.addObject("reason", arg.asElement());
-            }
-            else if (arg.matchesPrefix("targets", "target", "players")
-                    || arg.matchesArgumentList(PlayerTag.class)) {
-                scriptEntry.addObject("targets", arg.asType(ListTag.class).filter(PlayerTag.class, scriptEntry));
-            }
-        }
-        scriptEntry.defaultObject("reason", new ElementTag("Kicked."));
-        if (!scriptEntry.hasObject("targets")) {
-            throw new InvalidArgumentsException("Must specify target(s).");
-        }
-    }
-
-    @Override
-    public void execute(ScriptEntry scriptEntry) {
-        ElementTag reason = scriptEntry.getElement("reason");
-        List<PlayerTag> targets = (List<PlayerTag>) scriptEntry.getObject("targets");
-        if (scriptEntry.dbCallShouldDebug()) {
-            Debug.report(scriptEntry, getName(), db("targets", targets), reason);
-        }
+    public static void autoExecute(@ArgName("targets") @ArgLinear @ArgSubType(PlayerTag.class) List<PlayerTag> targets,
+                                   @ArgName("reason") @ArgPrefixed @ArgDefaultText("Kicked.") String reason) {
         for (PlayerTag player : targets) {
             if (player.isValid() && player.isOnline()) {
-                player.getPlayerEntity().kickPlayer(reason.toString());
+                if (Denizen.supportsPaper) {
+                    PaperAPITools.instance.kickPlayer(player.getPlayerEntity(), reason);
+                    return;
+                }
+                player.getPlayerEntity().kickPlayer(reason);

```

Comment:
No need for the check here, can just call `PaperAPITools#kickPlayer` as it has both a default implementation (which will be used on Spigot servers) and a Paper implementation